src/status_im/ui/screens/profile/user/views.cljs

```diff
@@ -25,84 +29,109 @@
+         (animation/start
+          (animation/timing
+           anim-opacity
+           {:toValue 0
+            :duration 200
+            :easing (.-ease (animation/easing))
+            :useNativeDriver true})
+          #(reset! to-hide false))))}
+    ;; TODO - change this to better
+    ;; This is really weird!
+    ;; We show this component whether or not @show is true
+    ;; Why use it at all?
+    ;; because we HAVE to deref `show` in render method for this to work
+    (when true
+      @show?
+      [react/animated-view {:style {:flex           1
+                                    :flex-direction :row
+                                    :margin-left    -40
+                                    :align-items    :center
+                                    :align-self     :stretch
+                                    :opacity        anim-opacity}}
+       [photos/photo photo-path {:size 40}]
+       [react/text {:style {:typography  :title-bold
+                            :line-height 21
+                            :margin-left 16
+                            :margin-right 40
+                            :text-align  :left}}
+        (multiaccounts/displayed-name contact)]])))
 
-(defn qr-viewer-toolbar [label value]
-  [toolbar/toolbar nil
-   (toolbar/nav-button
-    (toolbar.actions/close toolbar.actions/default-handler))
-   [toolbar/content-title label]])
-
-(defn qr-code-share-button [value]
-  (let [link (universal-links/generate-link :user :external value)]
-    [button/button-with-icon
-     {:on-press            #(list-selection/open-share {:message link})
-      :label               (i18n/label :t/share-link)
-      :icon                :main-icons/share
-      :accessibility-label :share-my-contact-code-button
-      :style               styles/share-link-button}]))
-
-(defview qr-viewer []
-  (letsubs [{:keys [value contact]} [:qr-modal]
-            ttt-enabled? [:tribute-to-talk/enabled?]]
-    [react/view styles/qr-code-viewer
-     [status-bar/status-bar {:type :modal-white}]
-     [qr-viewer-toolbar (multiaccounts/displayed-name contact) value]
-     [qr-code-viewer/qr-code-viewer
-      (merge
-       {:style         styles/qr-code
-        :value         value
-        :hint          (i18n/label :t/qr-code-public-key-hint)
-        :legend        (str value)
-        :show-tribute-to-talk-warning? ttt-enabled?}
-       (when-not platform/desktop?
-         {:footer-button qr-code-share-button}))]]))
-
-(defn- show-qr [contact source value]
-  #(re-frame/dispatch [:navigate-to :profile-qr-viewer {:contact contact
-                                                        :source  source
-                                                        :value   value}]))
+(defn my-profile-toolbar [{:keys [public-key photo-path] :as multiaccount} show?]
+  [toolbar/toolbar
+   {:transparent? true
+    :style        {:z-index 100
+                   :elevation 11}}
+   nil
+   [my-profile-toolbar-content multiaccount show?]
```

Comment:
so here `[my-profile-toolbar-content multiaccount @show?]`

Reply:
tried it just now.. it does not work

Maybe I didn't understand what you meant but,

I think it does not work because again, the view `my-profile-toolbar-conten` must respond dynamically to changes in `show?` atom and the component updates as a result and triggers appropriate animation. For that the atom has to be deref'd in render method and the problem with that is that it does not serve any visual purpose inside render but only to trigger animation

So, I did something else, which is still weird but less so

Reply:
Also.. good thing.. this interaction made me notice that I can change `:component-did-update` to `:component-will-update` which should theoretically make the animation a tad bit more responsive

so I changed it

Reply:
yeah maybe, because it should work, there are two way how update component, 1 - watch atom in render fn ,  2- watch params , and if you change show? param, not atom it should work, but ok ,we can fix it later

Reply:
Side note:

later on also.. I was thinking.. it will be good to do `:should-component-update` kind of optimizations - maybe not here, but in places like FlatList items so that they do not re-render unnecessarily.

I learn't about it from https://github.com/filipemerker/flatlist-performance-tips/ which I have bookmarked

Reply:
> yeah maybe, because it should work, there are two way how update component, 1 - watch atom in render fn , 2- watch params , and if you change show? param, not atom it should work, but ok ,we can fix it later

Understood.. I will study more about it. Definitely feels like we can do without resorting to weird hacks in situation like this.

Reply:
Oh.. 🤦‍♂ You are absolutely right

I know why it did not work, it is because I'm was not destructuring props in :component-will-update function

updating!

Reply:
done!

no more weird stuff :)